blocks/library/block/edit-panel/index.js

```diff
@@ -21,21 +21,31 @@ class SharedBlockEditPanel extends Component {
 		super( ...arguments );
 
 		this.bindTitleRef = this.bindTitleRef.bind( this );
+		this.bindEditButtonRef = this.bindEditButtonRef.bind( this );
 		this.handleFormSubmit = this.handleFormSubmit.bind( this );
 		this.handleTitleChange = this.handleTitleChange.bind( this );
 		this.handleTitleKeyDown = this.handleTitleKeyDown.bind( this );
 	}
 
-	componentDidMount() {
-		if ( this.props.isEditing ) {
+	componentDidUpdate( prevProps ) {
+		// Select the input text only once when the form opens.
+		if ( ! prevProps.isEditing && this.props.isEditing ) {
 			this.titleRef.select();
 		}
+		// Move focus back to the Edit button after pressing Save or Cancel.
+		if ( ! this.props.isEditing && ! this.props.isSaving ) {
+			this.editButton.focus();
+		}
 	}
 
 	bindTitleRef( ref ) {
 		this.titleRef = ref;
 	}
 
+	bindEditButtonRef( ref ) {
```

Comment:
Aside: React 16.3.0 introduced a new `createRef` function, which is arguably easier to use because it doesn't require this callback function, only to assign the ref in `constructor` and use it in `render`: 

https://reactjs.org/docs/refs-and-the-dom.html#creating-refs

Main difference is that it has to be accessed by the `.current` property.

Would introduce a bit of inconsistency in this component, but I'd argue it's an incremental step in the right direction.

Reply:
> Would introduce a bit of inconsistency
I can change also the other one, though unrelated.